src/repositories/courses-repository.js

```diff
@@ -37,124 +38,25 @@ module.exports = function createCoursesRepository(sequelize) {
       },
     };
 
-    if (specialtyId) {
-      return Course.findAndCountAll({
-        ...filter,
-        raw: true,
-        subQuery: false,
-        include: [{
-          model: Specialty,
-          required: true,
-          where: {
-            id: specialtyId,
-          },
-        }],
-      });
-    }
+    let response = null;
 
-    if (groupId) {
-      return Course.findAndCountAll({
-        ...filter,
-        raw: true,
-        subQuery: false,
-        include: [{
-          model: Specialty,
-          required: true,
-          include: [{
-            model: Group,
-            required: true,
-            where: {
-              id: groupId,
-            },
-          }],
-        }],
-      });
-    }
+    const modelsCollection1 = [Specialty];
+    const modelsCollection2 = modelsCollection1.concat([Group]);
+    const modelsCollection3 = modelsCollection2.concat([Student]);
+    const modelsCollection4 = [Teacher];
+    const modelsCollection5 = modelsCollection4.concat([LabTask]);
+    const modelsCollection6 = modelsCollection5.concat([LabReport]);
 
-    if (studentId) {
-      return Course.findAndCountAll({
-        ...filter,
-        raw: true,
-        subQuery: false,
-        include: [{
-          model: Specialty,
-          required: true,
-          include: [{
-            model: Group,
-            required: true,
-            include: [{
-              model: Student,
-              required: true,
-              where: {
-                id: studentId,
-              },
-            }],
-          }],
-        }],
-      });
-    }
+    response = handleId(specialtyId, response, Course, filter, modelsCollection1);

```

Comment:
Good job on creating a single function `handleId`, but I think you can go further by having a single function call for the handleId function,

You can get the param key by using `Object.keys(req.query.params)` and you can create a switch function to get the list of Models by the key name;

```
function getModels(key) {
   switch(key) {
      case 'studentId': return [Specialty, Group, Student]; //
   }
}
```